lib/src/easy_localization_app.dart

```diff
@@ -0,0 +1,146 @@
+        .dependOnInheritedWidgetOfExactType<_EasyLocalizationProvider>();
+  }
+
+  @override
+  bool updateShouldNotify(_EasyLocalizationProvider oldWidget) => true;
+}
+
+class EasyLocalization extends StatefulWidget {
+  final Widget child;
+  final List<Locale> supportedLocales;
+  final Locale fallbackLocale;
+  final _EasyLocalizationDelegate delegate;
+  final bool useOnlyLangCode;
+  final String path;
+  final AssetLoader assetLoader;
+
+  EasyLocalization({
+    @required this.child,
+    @required this.supportedLocales,
+    @required this.fallbackLocale,
+    @required this.path,
+    this.useOnlyLangCode = false,
+    this.assetLoader = const RootBundleAssetLoader(),
+  })  : assert(supportedLocales.contains(fallbackLocale)),
+        delegate = _EasyLocalizationDelegate(
+          path: path,
+        );
+
+  _EasyLocalizationState createState() => _EasyLocalizationState();
+
+  static _EasyLocalizationState of(BuildContext context) =>
+      _EasyLocalizationProvider.of(context).data;
+}
+
+class _EasyLocalizationState extends State<EasyLocalization> {
+  Locale _locale;
+  SharedPreferences _preferences;
+  Locale get locale => _locale;
+  List<Locale> get supportedLocales => widget.supportedLocales;
+  _EasyLocalizationDelegate get delegate => widget.delegate;
+
+  @override
+  void initState() {
+    super.initState();
+    changeLocale();
+  }
+
+  // locale is either the deviceLocale or the MaterialApp widget locale.
+  // This function is responsible for returning a locale that is supported by your app
+  // if the app is opened for the first time and we only have the deviceLocale information.
+  Locale localeResolutionCallback(
+      Locale locale, Iterable<Locale> supportedLocales) {
+    if (supportedLocales != widget.supportedLocales)
+      throw new Exception(
+          "You haven't given EasyLocalization.supportedLocales to MaterialApp ssupportedLocales property");
+
+    if (supportedLocales.contains(locale)) return locale;
+
+    return widget.fallbackLocale;
+  }
+
+  void changeLocale({Locale locale}) async {
+    var _defaultLocal;
+    print(
+        "================== changeLocale ${locale?.toLanguageTag()}===================");
+    try {
+      _preferences = await SharedPreferences.getInstance();
+      if (locale == null) {
+        var _codeLang = _preferences.getString('codeLa');
+        var _codeCoun = _preferences.getString('codeCa');
+        if (_codeLang == null) {
+          var currentLocale = Intl.getCurrentLocale().split("_");
+          _defaultLocal = Locale(currentLocale[0], currentLocale[1]);
+        } else {
+          _defaultLocal = Locale(_codeLang, _codeCoun);
+        }
+      } else
+        _defaultLocal = locale;
+      await _preferences.setString('codeCa', _defaultLocal.countryCode);
+      await _preferences.setString('codeLa', _defaultLocal.languageCode);
+
+      print(_defaultLocal.toString());
+
+      setState(() {
+        _locale = _defaultLocal;
+      });
```

Comment:
I think you missed the last commits, as https://github.com/aissat/easy_localization/commit/0d3c933490351842674d82772f7886fb95b5bdd3 changes `Intl.defaultLocale` here